### PR TITLE
PFC-BG_model implemented in pyTorch 

### PR DESCRIPTION
**Pull request by dylanhxzhao**
These changes have implemented the PFC-BG model into pyTorch. This is based on the manuscript for: 

Blanco-Pozo, M., Akam, T., & Walton, M. (2023). Dopamine-independent state inference mediates expert reward guided decision making bioRxiv, 2023-04.

The model consists of a recurrent neural network representing prefrontal cortex (PFC) and a feedforward network representing basal-ganglia. The PFC network is trained to predict the next observation and in doing so learns to infer hidden task states. The basal ganglia network is trained using actor-critic RL (A2C) to predict future reward and choose appropriate actions given the current observation and PFC activity.

### CODE REVIEW

**Comment by ThomasAkam** [comment]
Massive thanks @dylanhxzhao for implementing this pyTorch version of the modelling.
